### DISCUSSION TITLE
Ensemble Programming Workshop for Scripting Documentation Development on this Saturday

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#516] [Announcements]
I invite all interested people to a workshop announced during the recent Freeplane conference. I'll give a short introduction to Ensemble Programming also known as Mob Programming - a technique usually used for software development in a team.  And we apply this technique and related free online tools for writing documentation about Scripting in Freeplane together.

There is no registration, but if you are going to come, please leave a comment below.

We meet at our exclusive online location https://spatial.chat/s/freeplane .

See you!
Dimitry

[EnsembleProgrammingWorkshop.ics.zip](https://github.com/freeplane/freeplane/files/9226729/EnsembleProgrammingWorkshop.ics.zip)

Time: Saturday, August 13, 2022
13:00 London
14:00 Amsterdam, Berlin, Rome, Stockholm, Vienna
online, https://spatial.chat/s/freeplane

**Comment by euu2021** [comment]
Interesting! I will check in and try to help.

**Comment by EdoFro** [comment]
I will try too. 

Are there any particular software we have to have installed in our computers for this workshop?

thanks in advance, bye
edo

**Comment by sifran-github** [comment]
i will try to join.

**Comment by macmarrum** [comment]
I'll be there

**Comment by EdoFro** [comment]
finally I can’t take part tomorrow.
have fun 
bye,
edo

**Comment by macmarrum** [comment]
@dpolivaev many thanks for hosting the workshop. Ensemble programming is very interesting.

In addition, it was great to see Gitpod in action. The ability to have a virtual machine in the cloud (no need to install anything on my workstation), with VS Code in the browser, is amazing.

It's good to know which tools (like the on-line timer and the `mob` command-line utility you showed us) can be used to make ensemble programming smoother.

:fireworks:

**Comment by nnako** [comment]
Hi,

unfortunately, I was not able to join the meeting, although I would have been interested to learn about the technology. Has there been a audio / screen share recording? Otherwise, @macmarrum mentioned some tools (Gitpod, Virtual Machines and VisualStudio Code) which seem to belong to a greater tool chain used for the purpose of Ensemble Programming. 

(1) Would it be possible to sketch the entire tool chain, here? So, people interested might get an idea and a starting point for their own contributions.

(2) Would this tool chain be restricted to the purpose of Ensemble Programming, and in particular, for creation of documentations? Or would it be usable also for script development itself and maybe even Java programming for Freeplane?

Thanks.
Nnako

**Reply by dpolivaev** [reply]
I prepared written instructions at https://docs.google.com/document/d/1MrbspzMffyHxYaFmyYtioOg4addCB7fe3JBlNDxf0C0
You can have a look.
Obviously, the mentioned access token is no longer valid.
